src/Microsoft.AspNetCore.Routing/EndpointDataSource.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    public abstract class EndpointDataSource
+    {
+        public abstract IChangeToken ChangeToken { get; }
```

Comment:
what is the scenario behind having this change token here?

Reply:
Route data and endpoints can change after startup. MVC already has this feature 😆 - if you drop in a Razor Page while the site is running, it will be added to the route table.